### PR TITLE
[backend] refactor: extract service instance methods in app layer (#1115)

### PR DESCRIPTION
**Pull request by Kroustille**
# Context
> Briefly describe the purpose and context of this PR. Include relevant screenshots or screen recordings for the product team.

Refactors the backend `services` GraphQL resolvers to delegate more service-instance-related logic to the `ServiceInstanceApp` application layer, and hardens some domain queries by switching to parameterized SQL.

**Changes:**

- Move SEO/link/subscribed service-instance query formatting (global IDs, NotFound handling) into `ServiceInstanceApp`.
- Parameterize raw SQL fragments in `service-instance.domain.ts` to avoid string interpolation.
- Update/add Vitest coverage for the refactored app-layer methods.

## Related issues

- Related to #1115 

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [x] I tested all features and edge cases
- [x] I refactored code where necessary to improve quality
- [x] I made sure my code meets development guidelines
- [x] I performed a self-review of my code

## Testing

- [x] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [ ] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [x] Unit tests
- [x] Integration tests
- [ ] E2E tests
- [x] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/portal-api/src/modules/services/service-instance.app.test.ts [hunk 1]
```diff
@@ -283,391 +294,655 @@ describe('Service Instance app', () => {
...
+            serviceInstanceId: toGlobalId('ServiceInstance', mockId),
+            name: 'Name',
+          },
+          null
+        )
+      ).rejects.toThrow('SERVICE_DEFINITION_NOT_FOUND');
+    });
+
+    it('should throw when user cannot modify platform service', async () => {
+      // Cannot test with real DB: the bypass user always passes
```
Comment (carinelebas) [inline apps/portal-api/src/modules/services/service-instance.app.test.ts]: Maybe you could test by changing the user in the context if the issue is that test is executed with bypass user?
